db/db_iter.cc

```diff
@@ -753,13 +761,13 @@ bool DBIter::FindValueForCurrentKey() {
       if (last_not_merge_type == kTypeDeletion ||
           last_not_merge_type == kTypeSingleDeletion ||
           last_not_merge_type == kTypeRangeDeletion) {
-        MergeHelper::TimedFullMerge(merge_operator_, saved_key_.GetKey(),
+        s = MergeHelper::TimedFullMerge(merge_operator_, saved_key_.GetKey(),
```

Comment:
My question is when we encounter error for, say key1, then iterate to key2 without error, should the iterator stay in error state or okay state? Seems with existing logic we'll stay in error state, but I actually prefer the latter.